Neos.Fusion/Classes/FusionObjects/ActionUriImplementation.php

```diff
@@ -195,13 +233,26 @@ public function evaluate()
         }
 
         try {
-            return $uriBuilder->uriFor(
+            $arguments = $this->getArguments();
+            $routingArguments = $this->getRoutingArguments();
+            if ($arguments && $routingArguments) {
+                throw new RuntimeException('Neos.Fusion:ActionUri does not allow to combine "arguments" and "routingArguments"', 1665431866);
+            }
+            $uriString = $uriBuilder->uriFor(
                 $this->getAction(),
-                $this->getArguments(),
+                $routingArguments ?: $arguments ?: [],
                 $this->getController(),
                 $this->getPackage(),
                 $this->getSubpackage()
             );
+            $queryParameters = $this->getQueryParameters();
+            if (empty($queryParameters)) {
+                return $uriString;
+            }
+            $uri = new Uri($uriString);
```

Comment:
That would look great in a flow uri utility or service, but I know the hassle since it’s in another repo… it might be enough time to pull it up (and you have my +1)


Advantage beeign that we can super easily tests this ugly code snippet isolated against all scenarios.

Reply:
Eventually this code will be replaced by an all new action-uri-builder on the flow side. See https://github.com/neos/flow-development-collection/pull/2744 (not finished and still in discussion. Once this is in place ActionUri should be adjusted to use this internally. 

In the meantime this pr allows to handle the negative effects of additionalParams ending up in the routingCache and introduces the distinction between `routingArguments` and `queryParameters`.

Reply:
Okay i see then its totally fine with inlining this temporarily ;)

Reply:
FYI now with https://github.com/neos/flow-development-collection/pull/3316 in place we can actually use a proper helper instead ;)